test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,552 @@
+        await task.reload();
+        expect(task.ProjectId).to.not.be.null;
+      });
+    });
+
+    describe('explicitly validating primary/auto incremented columns', async () => {
+      it('should emit an error when we try to enter in a string for the id key without validation arguments', async () => {
+        class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+          declare id: number | string;
+        }
+
+        User.init({
+          id: {
+            type: DataTypes.INTEGER,
+            autoIncrement: true,
+            primaryKey: true,
+            validate: {
+              isInt: true,
+            },
+          },
+        }, { sequelize });
+
+        await User.sync({ force: true });
+
+        try {
+          await User.create({ id: 'helloworld' });
+        } catch (error) {
+          assert(error instanceof Error);
+          if (error instanceof ValidationError) {
+            expect(error.get('id')[0].message).to.equal('Validation isInt on id failed');
+          }
+        }
+      });
+
+      it('should emit an error when we try to enter in a string for an auto increment key (not named id)', async () => {
+        class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+          declare username: string;
+        }
+
+        User.init({
+          username: {
+            type: DataTypes.INTEGER,
+            autoIncrement: true,
+            primaryKey: true,
+            validate: {
+              isInt: {
+                msg: 'Username must be an integer!',
+              },
+            },
+          },
+        }, { sequelize });
+
+        await User.sync({ force: true });
+
+        try {
+          await User.create({ username: 'helloworldhelloworld' });
+        } catch (error) {
+          assert(error instanceof ValidationError);
+          expect(error.get('username')[0].message).to.equal('Username must be an integer!');
+        }
+      });
+
+      describe('primaryKey with the name as id with arguments for its validation', async () => {
+        disableDatabaseResetForSuite();
+        const vars = beforeAll2(async () => {
+          class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+            declare id: number | string;
+          }
+
+          User.init({
+            id: {
+              type: DataTypes.INTEGER,
+              autoIncrement: true,
+              primaryKey: true,
+              validate: {
+                isInt: {
+                  msg: 'ID must be an integer!',
+                },
+              },
+            },
+          }, { sequelize });
+
+          await User.sync({ force: true });
+
+          return { User };
+        });
+
+        it('should emit an error when we try to enter in a string for the id key with validation arguments', async () => {
+          try {
+            await vars.User.create({ id: 'helloworld' });
+          } catch (error) {
```

Comment:
You still have a few tests that use try-catch instead of `expect(x).to.be.rejected`, can you update all of them?